lib/vanagon/component.rb

```diff
@@ -290,6 +290,14 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      obj = @options
+      [:version, :name].each do |var|
+        obj[var.to_s] = self.instance_variable_get("@#{var}")
```

Comment:
Isn't this going to modify the `@options` variable?  Is that intended?

Also this doesn't seem necessary to metaprogram since it's just 2 variables.

I'd rather just be direct and do:
```ruby
obj = @options
obj['version'] = @version
obj['name'] = @name
{@name => obj}
```

But then the `obj` variable seems unnecessary, so I'd consider just doing this:
```ruby
@options['version'] = @version
@options['name'] = @name
{@name => @options}
```

The `obj` only seems necessary if we're going to make a clone of `@options` so we don't modify it here.